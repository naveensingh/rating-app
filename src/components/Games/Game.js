import React from 'react';
import PropTypes from 'prop-types';
import './games.scss';
import { useDispatch } from 'react-redux';
import Rating from '../Rating/Rating';

const Game = ({ game }) => {
  const { uuid, title, description, rating, avatar } = game;
  const dispatch = useDispatch();
  const avatarPath = require(`../../static/images/${avatar}`);
  const setRating = ratingValue => dispatch({
    type: 'SET_RATING',
    payload: {
      uuid,
      title,
      description,
      avatar,
      rating: ratingValue
    }
  });
  return (
    <div className="game" key={uuid}>
      <div className="game--avatar">
        <img src={avatarPath} alt={title} />
      </div>
      <div className="game--meta">
        <div className="game--title">
          {title}
        </div>
        <div className="game--description">
          {description}
        </div>
        <div className="game--rating">
          <Rating data-testid="rating" rating={rating} onClick={setRating} />
        </div>
      </div>
    </div>
  );
};

Game.propTypes = {
  game: PropTypes.shape({
    uuid: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
  }).isRequired
};

export default Game;
