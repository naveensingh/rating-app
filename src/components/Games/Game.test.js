import React from 'react';
import Game from "./Game";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {shallow} from "enzyme";

const store = configureMockStore([])({});

const games = {
    "title": "Game",
    "avatar": "CallOfDuty.jpeg",
    "description": "Lorem ipsum dolor sit amet",
    "rating": 0,
    "uuid": "someGame"
};

describe('Games spec', () => {
    test('should match snapshot for available games', () => {
        const gamesComponent = shallow(
            <Provider store={store}>
                <Game game={games}/>
            </Provider>);
        expect(gamesComponent).toMatchSnapshot();
    });
});
