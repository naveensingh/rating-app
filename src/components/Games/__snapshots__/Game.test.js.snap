// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games spec should match snapshot for available games 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Game
    game={
      Object {
        "avatar": "CallOfDuty.jpeg",
        "description": "Lorem ipsum dolor sit amet",
        "rating": 0,
        "title": "Game",
        "uuid": "someGame",
      }
    }
  />
</ContextProvider>
`;
